colors/primary.vim

```diff
@@ -93,49 +94,56 @@ endif
 
 
 " Colors for Syntax Highlighting.
-exe 'hi String       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'='.s:I
-exe 'hi Character    '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'='.s:I
-exe 'hi Conditional  '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-exe 'hi Label        '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-exe 'hi Repeat       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-exe 'hi Statement    '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-exe 'hi Keyword      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-exe 'hi Exception    '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:RED  .' '.s:M.'=bold'
-
-exe 'hi Normal       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:BLUE .' '.s:M.'=none'
-exe 'hi Identifier   '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:BLUE .' '.s:M.'=bold'
-exe 'hi Function     '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:BLUE .' '.s:M.'=bold'
-
-exe 'hi Comment      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=none'
-exe 'hi Typedef      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'='.s:I
-exe 'hi PreProc      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-exe 'hi Include      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-exe 'hi Define       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-exe 'hi Macro        '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-exe 'hi Precondit    '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-exe 'hi SpecialComment '.s:M.'bg='.s:BG.' '.s:M.'fg='.s:GREEN.' '.s:M.'=bold'
-
-exe 'hi Special      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY0.' '.s:M.'=none'
-exe 'hi Delimiter    '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY0.' '.s:M.'=none'
-exe 'hi Debug        '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY0.' '.s:M.'=none'
-exe 'hi SpecialChar  '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY0.' '.s:M.'=bold'
-exe 'hi Title        '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY0.' '.s:M.'=bold'
-
-exe 'hi Constant     '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Boolean      '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Number       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Float        '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Operator     '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
-exe 'hi Tag          '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
-exe 'hi Ignore       '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1.' '.s:M.'=none'
-exe 'hi Underlined   '.s:M.'bg='.s:BG  .' '.s:M.'fg='.s:GREY1 .' '.s:M.'=underline'
+if (s:enable_bg_transparency == 1)
+  exe 'hi Normal       '.s:M.'bg=NONE '.s:M.'fg='.s:BLUE .' '.s:M.'=none'
+  exe 'hi NonText      '.s:M.'bg=NONE '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
+else
+  exe 'hi Normal       '.s:M.'bg='.s:BG .' '.s:M.'fg='.s:BLUE .' '.s:M.'=none'
+  exe 'hi NonText      '.s:M.'bg='.s:BG   .' '.s:M.'fg='.s:GREY1.' '.s:M.'=bold'
```

Comment:
Thanks for contributing! Looks like there are some whitespace inconsistencies on lines 98-102. Please fix.

Reply:
Hi, I already fixed it.

Reply:
I didn't see a change here. Strange spacing still on `'bg='.s:BG   .' '` here and farther down on the `hi Underlined` line.

Sorry for the back-and-forth. If it's a pain to update now, I might do a pass over the file later to make a few overall readability tweaks.

Reply:
I've formatted the file globally to make it more readable, and thanks for your patience:)

> I didn't see a change here. Strange spacing still on `'bg='.s:BG .' '` here and farther down on the `hi Underlined` line.
> 
> Sorry for the back-and-forth. If it's a pain to update now, I might do a pass over the file later to make a few overall readability tweaks.

Reply:
Looks good to me!